pkg/thanos/statefulset.go

```diff
@@ -301,6 +301,20 @@ func makeStatefulSetSpec(tr *monitoringv1.ThanosRuler, config Config, ruleConfig
 		trCLIArgs = append(trCLIArgs, fmt.Sprintf("--alert.query-url=%s", tr.Spec.AlertQueryURL))
 	}
 
+	if tr.Spec.AlertRelabelConfig != nil {
```

Comment:
`--alert.relabel-config` and `--alert.relabel-config-file` are mutually exclusive (see [here](https://thanos.io/v0.23/components/rule.md/#flags)) so the operator should do

```
if tr.Spec.AlertRelabelConfigFile != nil {
    ...
} else if tr.Spec.AlertRelabelConfig != nil {
    ...
}
```

BTW the same is true for `-objstore.config` and `-objstore.config-file` so it would be nice if you could fix it too :)